src/frontend/src/components/FilterPanel/components/DateFilter/index.tsx

```diff
@@ -22,40 +22,44 @@ import {
 
 export type DateType = FormattedDateType | Date;
 
+const formatDate = (d: DateType) => {
+  if (d === undefined) return d;
+  if (typeof d === "string") return d;
+
+  return d.toISOString().substring(0, 10);
+};
+
+export interface DateMenuOption {
+  name: string; // Must be UNIQUE because we assume can be used as key/id
+  // For both `numDays...` it's relative to now() when option is chosen.
+  // Assume start is guaranteed, but end cap is not.
+  numDaysEndOffset?: number; // How far back end of date interval is
+  numDaysStartOffset: number; // How far back start of date interval is
+}
+
 interface Props {
   fieldKeyEnd: string;
   fieldKeyStart: string;
   inputLabel: string;
   updateDateFilter: (start: FormattedDateType, end: FormattedDateType) => void;
+  menuOptions: DateMenuOption[];
```

Comment:
this is great

Reply:
like that you added this prop and now i feel like this is all much more organized ❤️